lib/hooks/useDocumentsLoader.ts

```diff
@@ -0,0 +1,100 @@
+import { SupabaseClient } from "@supabase/supabase-js";
+import { useEffect, useState } from "react";
+import { Database } from "../database.types";
+import { DocChunk } from "../types";
+
+export type Document = {
+  docChunks: DocChunk[];
+  pageName: string;
+  sectionName: string;
+  createdAt: string;
+  url: string;
+};
+
+export default function useDocumentsLoader(supabase: SupabaseClient<Database>) {
+  const PAGE_SIZE = 10;
+
+  const [allDocumentCount, setAllDocumentCount] = useState(0);
+  const [docPage, setDocPage] = useState<number>(1);
+  const [docs, setDocs] = useState<Document[]>([]);
+
+  const fetchAllSectionCount = async () => {
+    const { data } = await supabase.rpc("get_all_page_section_counts");
+    if (data) {
+      setAllDocumentCount(data);
+    }
+  };
+
+  const deleteDocument = async (document: Document) => {
+    const documentChunkIds = document.docChunks.map((docChunk) =>
+      docChunk.id.toString(),
+    );
```

Comment:
Think this `.toString()` is unnecessary - think it was a suggestion I made which was wrong!
```suggestion
    const documentChunkIds = document.docChunks.map((docChunk) => docChunk.id);
```